app/backend/src/clj/teet/thk/thk_mapping.clj

```diff
@@ -24,6 +25,24 @@
           msb (.longValue (.shiftRight n 64))]
       (java.util.UUID. msb lsb))))
 
+(defonce random (java.security.SecureRandom.))
+
+(defn random-small-uuid
+  "Create a UUID that only has 64bits number. This is because THK
+  teet id numbers are only int8 sized and cannot fit a full UUID"
+  []
+  (number->uuid (.nextLong random)))
```

Comment:
A problem with the default Java SecureRandom used to be that it could hang for long times because it had a bad default randomness source on Linux (worked around by starting JVM with  -Djava.security.egd=file:/dev/urandom ). Hopefully this is fixed the JVM version used by Datomic...